fw/rbcx-coprocessor/include/Motor.hpp

```diff
@@ -3,27 +3,77 @@
 #include "utils/Debug.hpp"
 #include "utils/Regulator.hpp"
 
-#include <algorithm>
 #include <stdint.h>
 
 inline const uint16_t motorLoopFreq = 100;
 
 class Motor {
 public:
-    Motor()
-        : m_velocityReg(INT16_MAX, 150000, 300000, 20000)
-        , m_positionReg(INT16_MAX, 150000, 300000, 20000)
-        , m_state(PowerCtrl) {}
-
-    std::pair<int16_t, bool> poll(uint16_t encTicks) {
-        int16_t actualTicksPerLoop = encTicks - m_lastEncTicks;
-        m_actualPosition += actualTicksPerLoop;
+    Motor() { m_lastEncTicks = 0; }
+
+    void reset() {
+        m_velocityReg = Regulator(INT16_MAX, 150000, 300000, 20000);
+        m_positionReg = Regulator(500, 1000, 0, 0);
+        m_dither = 0;
+        m_targetVelocity = 0;
+        m_actualPower = 0;
+        m_actualPosition = 0;
+        m_targetPosition = 0;
+        m_actualTicksPerLoop = 0;
+        m_posEpsilon = 3;
+        m_velEpsilon = 3;
+        m_maxAccel = 2000 / motorLoopFreq;
+        m_mode = MotorMode_POWER;
+    }
```

Comment:
Necheme volat `reset()` z konstruktoru?

Reply:
Lol jsem myslel, že jsem to tam napsal :'D spraveno